core/server/errors/index.js

```diff
@@ -34,6 +34,28 @@ function getConfigModule() {
     return config;
 }
 
+function isValidErrorStatus(status) {
+    return _.isNumber(status) && status >= 400 && status < 600;
+}
+
+function getStatusCode(error) {
+    if (error.statusCode) {
+        return error.statusCode;
+    }
+
+    if (error.status && isValidErrorStatus(error.status)) {
+        error.statusCode = error.status;
+        return error.statusCode;
+    }
+
+    if (error.code && isValidErrorStatus(error.code)) {
+        error.statusCode = error.code;
+        return error.statusCode;
+    }
+
+    return 500;
```

Comment:
all the other branches of this function ensure that the `error` has a `statusCode` property before returning. Should this final branch do the same?